src/styled.js

```diff
@@ -1,29 +1,37 @@
 import { css } from './css';
 
-let h;
-const setPragma = pragma => (h = pragma);
+let h, forwardRef, useTheme;
+const setup = (_h, _forwardRef, _useTheme = () => {}) => {
+    h = _h;
+    forwardRef = _forwardRef;
+    useTheme = _useTheme;
+};
 
 /**
  * Styled function
  * @param {String} tag
  */
 function styled(tag) {
     const _ctx = this || {};
-
     return function() {
         const _args = arguments;
-
-        return function Styled(props) {
-            const _props = (_ctx.p = Object.assign({}, props));
+        function Styled(props, ref) {
+            const theme = useTheme();
```

Comment:
Does it has to be a function? Could it be a plain object? Could become a perf bottleneck on user land.

Reply:
This is to get feature parity with other CSS-in-JS theming capability, i.e.
```jsx
const useTheme = () => useContext(ThemeContext)
```

Here's how it's similarly done in `styled-components@5` - https://github.com/styled-components/styled-components/blob/v5/packages/styled-components/src/models/StyledComponent.js#L129

Reply:
For non context user, it could be just `() => theme`